dashboard/test/controllers/api/v1/pd/application/teacher_applications_controller_test.rb

```diff
@@ -80,31 +72,25 @@ class TeacherApplicationsControllerTest < ::ActionController::TestCase
       response: :forbidden
 
     test 'sends email on successful create' do
-      Pd::Application::TeacherApplicationMailer.expects(:confirmation).
-        with(instance_of(TEACHER_APPLICATION_CLASS)).
-        returns(mock {|mail| mail.expects(:deliver_now)})
-
       sign_in @applicant
 
       put :create, params: @test_params
+      application = TEACHER_APPLICATION_CLASS.last
+      assert_equal 1, application.emails.where.not(sent_at: nil).where(email_type: 'confirmation').count
```

Comment:
this could be hard to debug if the create fails. I would suggest leaving the `assert_response :success` right after the `put`, here and below

Reply:
Good call!